include/hook.h

```diff
@@ -109,6 +107,15 @@ typedef struct
 	const char *error;
 } hook_data_channel_approval;
 
+typedef struct
+{
+	struct Client *client;
+	struct Channel *chptr;
+	struct membership *clientms;
+	struct membership *targms;
```

Comment:
Let's not abbreviate to save 2 characters

```suggestion
	struct membership *targetms;
```